zebrad/src/components/inbound.rs

```diff
@@ -360,22 +375,36 @@ impl Service<zn::Request> for Inbound {
                     })
                     .try_collect::<Vec<_>>()
                     .map_ok(|blocks| {
-                        if blocks.is_empty() {
-                            zn::Response::Nil
-                        } else {
-                            zn::Response::Blocks(blocks)
-                        }
+                        // Work out which hashes were missing.
+                        let available_hashes: HashSet<block::Hash> = blocks.iter().map(|block| block.hash()).collect();
+                        let available = blocks.into_iter().map(Available);
+                        let missing = hashes.into_iter().filter(|hash| !available_hashes.contains(hash)).map(Missing);
+
+                        zn::Response::Blocks(available.chain(missing).collect())
```

Comment:
I still need to write tests for this, but I think they should be pretty simple.

Do we have any tests that run against a full Zebra inbound service, network stack, and empty state?

(There shouldn't be any configured peers, so the state stays empty. But I can fix that in my test setup.)

Reply:
This seems like a good place to start:
https://github.com/ZcashFoundation/zebra/blob/1a14baf8e7c6c82dba7a64e9b43d076b8f856f2a/zebrad/src/components/inbound/tests.rs#L655